src/libpriv/rpmostree-core.c

```diff
@@ -1353,39 +1350,23 @@ rpmostree_context_prepare_install (RpmOstreeContext    *self,
  */
 void
 rpmostree_dnf_add_checksum_goal (GChecksum   *checksum,
-                                 HyGoal       goal,
-                                 GHashTable  *pkg_to_checksum)
+                                 HyGoal       goal)
 {
   g_autoptr(GPtrArray) pkglist = NULL;
-  g_autoptr(GPtrArray) pkg_checksums =
-    g_ptr_array_new_with_free_func (g_free);
+  guint i;
+  g_autoptr(GPtrArray) pkg_checksums = g_ptr_array_new_with_free_func (g_free);
 
   pkglist = hy_goal_list_installs (goal, NULL);
   g_assert (pkglist);
-  for (guint i = 0; i < pkglist->len; i++)
+  for (i = 0; i < pkglist->len; i++)
     {
       DnfPackage *pkg = pkglist->pdata[i];
+      int pkg_checksum_type;
+      const unsigned char *pkg_checksum_bytes = dnf_package_get_chksum (pkg, &pkg_checksum_type);
+      g_autofree char *pkg_checksum = hy_chksum_str (pkg_checksum_bytes, pkg_checksum_type);
+      char *pkg_checksum_with_type = g_strconcat (hy_chksum_name (pkg_checksum_type), ":", pkg_checksum, NULL);
```

Comment:
Yeah, I definitely agree. Though I left it as is -- that fixup is actually just restoring code that my initial commits replaced. I opted to just let it be to minimize the diff and make git blame less noisy.